lib/lotus-client.js

```diff
@@ -0,0 +1,51 @@
+import pRetry from 'p-retry'
+import { rpc } from './rpc-client.js'
+/**
+ * @param {object} options
+ * @param {string} rpcUrl
+ * @param {string} rpcAuth
+ * @param {number} [options.maxAttempts]
+ * @param {function} [options.rpcFn]
+ * @returns {Promise<string>} The chain head Cid
+ */
+export async function getChainHead(rpcUrl, rpcAuth, { maxAttempts = 5, rpcFn = rpc } = {}) {
```

Comment:
Since we are already injecting `rpcFn`, let's use that mechanism instead of adding more rpc-related parameters.

Such a change will make the `rpcFn` parameter required.

```
export async function getChainHead({ maxAttempts = 5, rpcFn } = {}) {
```